app/src/main/java/com/hajduk/systems/prepareordermailing/view/ScanBarCode.kt

```diff
@@ -18,14 +19,18 @@ class ScanBarCode : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        DomainDataHolder.clearData()
+        initializeApp()
         if (AuthorizationHolder.areCredentialsInContext(this)) {
             setContentView(R.layout.activity_scan_bar_code)
             scanBarCodeButton.setOnClickListener { initializeBarCodeScanning() }
         } else {
             startActivity(Intent(this, InitializeAuthentication::class.java))
         }
+    }
 
+    private fun initializeApp() {
+        ThreadExceptionHandler.configure()
+        DomainDataHolder.clearData()
```

Comment:
a to się nie powinno czyścić za każdym razem jak wbijamy do index albo scancode?
wiesz tam będą realizować wiele orderów
a onCreate się robi przy starcie albo restarcie apki
https://developer.android.com/guide/components/activities/activity-lifecycle

Reply:
Wywaliłem index, scan bar code to pierwszy ekran, więc ti chyba jest ok